### DISCUSSION TITLE
Forward projection for CTAC issue - help welcomed!

### DISCUSSION THREAD

**Discussion by kmo49** [UCL/STIR#1414] [Q&A]
Hello! 

I am trying to forward project a three dimensional CT object to use for attenuation correction of my PET image. I am modeling 3D PET with 24 rings, maximum segment of 23. My resultant PET sinogram is 249x210x576.  210 aa bins and 249 td bins. 

When I am forward projecting the CT image, the sinogram looks correct on the first slice and the last slice, but proceeds to look like stacked CT images towards the middle. 

I am getting the same issue with the calculate_attenuation function. I included images below, and would appreicate any advice on how to fix this issue, thank you! 
<img width="259" alt="Screenshot 2024-04-28 at 7 54 10 PM" src="https://github.com/UCL/STIR/assets/148734272/ac70d3f3-55fa-43ff-9e20-c50b968d978c">
<img width="254" alt="Screenshot 2024-04-28 at 7 54 19 PM" src="https://github.com/UCL/STIR/assets/148734272/3af096a9-7b04-4698-992d-612cb5880d4d">
<img width="257" alt="Screenshot 2024-04-28 at 7 54 36 PM" src="https://github.com/UCL/STIR/assets/148734272/4f3b840d-9537-4db9-baa1-98637cf610a8">
<img width="257" alt="Screenshot 2024-04-28 at 7 54 49 PM" src="https://github.com/UCL/STIR/assets/148734272/6af23741-f6da-4a22-abb9-7f7619241cdd">
<img width="264" alt="Screenshot 2024-04-28 at 7 54 57 PM" src="https://github.com/UCL/STIR/assets/148734272/1e57f09e-762c-466d-a84e-66a5332e9e7a">

**Comment by KrisThielemans** [comment]
I suspect you are reading the data wrongly. Are you doing this without STIR, e.g. read it as a "stack of sinograms"? In that case, it is incorrect. STIR currently writes the data in a different order, check the .hs file. This is fairly difficult to read unfortunately without STIR, so I recommend using STIR python
```python
proj_data = stir.ProjData.read_from_file('bla.hs')
proj_data_arr = stirextra.to_numpy(proj_data)
```
this array *is* a stack of sinograms. First seg 0, then +1, then -1, etc, see https://github.com/UCL/STIR/blob/42aebde091618e3ab7d35770e8cc5293b995f428/src/include/stir/ProjData.h#L257-L263

or similar lines in SIRF python.
